src/re.rs

```diff
@@ -632,8 +632,8 @@ impl Regex {
         }
     }
 
-    #[doc(hidden)]
-    pub fn names_iter<'a>(&'a self) -> NamesIter<'a> {
+    /// Returns an iterator over the capture names.
+    pub fn names_iter<'r>(&'r self) -> NamesIter<'r> {
```

Comment:
sure, i like those better as well. makes the doc comments redundant, though, so i’ll improve those as well